tests/PHPUnit/Unit/DateTest.php

```diff
@@ -72,6 +72,39 @@ public function testYesterday()
         $this->assertEquals(strtotime(date("Y-m-d", strtotime('-1day')) . " 00:00:00"), $date->getTimestamp());
     }
 
+    /**
+     * create last week object check that timestamp is correct (midnight)
+     *
+     * @group Core
+     */
+    public function testLastWeek()
+    {
+        $date = Date::lastWeek();
+        $this->assertEquals(strtotime(date("Y-m-d", strtotime('-1 week')) . " 00:00:00"), $date->getTimestamp());
+    }
+
+    /**
+     * create last month object check that timestamp is correct (midnight)
+     *
+     * @group Core
+     */
+    public function testLastMonth()
+    {
+        $date = Date::lastMonth();
+        $this->assertEquals(strtotime(date("Y-m-d", strtotime('-1 month')) . " 00:00:00"), $date->getTimestamp());
+    }
+
+    /**
+     * create last year object check that timestamp is correct (midnight)
+     *
+     * @group Core
+     */
+    public function testLastYear()
+    {
+        $date = Date::lastYear();
+        $this->assertEquals(strtotime(date("Y-m-d", strtotime('-1 year')) . " 00:00:00"), $date->getTimestamp());
+    }
```

Comment:
I think you could test these more effectively by using `Date::$now`. Right now it looks like you're mostly duplicating the function logic in the tests?

Reply:
@diosmosis  I've improved the date tests (and also fixed `yesterdaySameTime`, that actually wasn't respecting the now timestamp for the returned time)